hmt_escrow/job.py

```diff
@@ -911,13 +940,24 @@ def _validate_credentials(self, **credentials) -> bool:
             bool: returns True if the calculated and the given address match.
 
         """
-        addr = credentials["gas_payer"]
-        priv_key = credentials["gas_payer_priv"]
+        addr_valid = False
+        gas_payer_addr = credentials["gas_payer"]
+        gas_payer_priv = credentials["gas_payer_priv"]
 
-        priv_key_bytes = decode_hex(priv_key)
-        pub_key = keys.PrivateKey(priv_key_bytes).public_key
-        calculated_addr = pub_key.to_checksum_address()
-        return Web3.toChecksumAddress(addr) == calculated_addr
+        addr_valid = self._eth_addr_valid(gas_payer_addr, gas_payer_priv)
+
+        if not multi_credentials:
+            return addr_valid
+
+        for eth_addr, priv_key in multi_credentials:
```

Comment:
Do we care if they're ALL valid?  Really just care about the one we're going to use, no?

Reply:
I think it's good practice to let the person know in the setup step if they've typo'd the private key for example.

Reply:
I agree, always good to give more info .. but will 1 bad key out of 10 block a deploy?

Reply:
Yea it will block. I'd still say it's still good since it'll log it. 

Also, it's not as if we constantly change keys so it should be really few times a year thing.

Reply:
Why not just log the bad ones and take out of the pool of available ones to deploy with; that way we learn about the problem, and don't break any deploys at the same time

Reply:
Yeah, maybe credentials validation makes more sense with the "main credentials". For the other ones we could fail by logging and raffle new ones.

Only thing I'm thinking is that is it more probable that we fuck up only 1 key out of 10 or it's something more systemic where we fuck up all the keys?

Just a bit weird to me to validate main credentials but not validating the rest of them.

Reply:
> Only thing I'm thinking is that is it more probable that we fuck up only 1 key out of 10 or it's something more systemic where we fuck up all the keys?

I would think the former, like via a human improper copy / paste.

I'd validate em all, if we could!  Do we need to differentiate main vs secondary?

Reply:
"I'd validate em all, if we could!" what do you mean? This does exactly that now.

Reply:
Oh good, thats what I was thinking.  Was thrown off by:

> Just a bit weird to me to validate main credentials but not validating the rest of them.

Reply:
Main reason I want to distinguish "credentials" from "multi-credentials" is to not introduce breaking changes. That way we can use it with and without multi-credentials.

Reply:
Oh I gotcha, that makes sense.  So validate all, if main fails, abort; if remainder fails, proceed but warn.. right?

Reply:
So when when we do this:
`job = Job(credentials, manifest, multi_credentials=multi_credentials)`
we will currently fail that if `multi_credentials` contains wrong eth_addr:priv_key pairs (IMO this is not an issue since we should anyways check carefully that we use correct eth_addr:priv_key pairs)

After we know all keys are valid, we first try to do the escrow creation with `credentials` (main). If that fails we start going through `multi_credentials` and raffle until we succeed. So we will keep new creds until we succeed.

Reply:
Yeah I'm still just saying that if `multi_credentials` has at least 1 working credential, we should use it and ignore the ones that aren't valid; and throw a warning that theres a bad credential in multi_creds

Reply:
So should `_validate_credentials` just check there's at least one valid credential in `multi_credentials` and return true?

Reply:
Probably just return a list of valid ones for use, warn on the bad ones.  Raise exception / return false if none?

Reply:
Lol, actually we don't even validate `multi_credentials` in `_validate_credentials` so wrong alert! Works exactly as you described.

Reply:
We will anyways catch those in `_create_escrow` phase. The only edge case would be if ALL the creds were falsy. I think we are nitpicking quite a lot here.

Reply:
Oh lol. yeah I'm not really too concerned with the code itself more the behavior of a single bad credential in the list shouldnt fail the entire deploy.  :)